ocaml/xapi/xapi_xenops.ml

```diff
@@ -3130,9 +3124,10 @@ let task_cancel ~__context ~self =
 
 let md_of_vusb ~__context ~self =
   let vm = Db.VUSB.get_VM ~__context ~self in
-  let vusb = Db.VUSB.get_record ~__context ~self in
-  let pusb =Db.PUSB.get_record ~__context ~self:vusb.API.vUSB_attached in
-  MD.of_vusb ~__context ~vm:(Db.VM.get_record ~__context ~self:vm) ~pusb:pusb
+  let usb_group = Db.VUSB.get_USB_group ~__context ~self in
+  let pusb = List.hd (Db.USB_group.get_PUSBs ~__context ~self:usb_group) in
```

Comment:
This one should be caught and an internal_error should be raised (as this is used in VUSB.unplug)

Reply:
updated.